lib/omniauth/strategy.rb

```diff
@@ -180,6 +180,8 @@ def call!(env) # rubocop:disable CyclomaticComplexity, PerceivedComplexity
         raise(error)
       end
 
+      warn_if_using_get
```

Comment:
Hmm, I see what you're saying, however it will only log in the case that `OmniAuth.config.allowed_request_methods.include?(:get)` resolves to true. (See line 198)

Reply:
Yeah, correct. During the upgrade process, we wanted to track down any potential cases where `GET` is used instead of the legit `POST`. We relied on the https://github.com/omniauth/omniauth/wiki/Upgrading-to-2.0#get that says

> Because using GET for login poses concerns (see OWASP Cheatsheet), after upgrading OmniAuth will log a :warn level log with **every GET request to a login path** when your OmniAuth.config.allowed_request_methods include :get.

Instead, we've uncovered that the warning is produced to any GET request to our app and the log stream is getting bloated pretty fast.

Should the `warn_if_using_get` be guarded with `on_auth_path?` check? We can work on the PR if you confirm.

Reply:
I think that would be a reasonable resolution, yes. A PR would be appreciated and I'll hopefully have some time soon to go through the release w/ a patch bump for you.

Reply:
@BobbyMcWho Hi, I've prepared a PR for this issue https://github.com/omniauth/omniauth/pull/1041